syntax/crystal.vim

```diff
@@ -288,7 +288,7 @@ if !exists('g:crystal_no_special_methods')
   syn match   crystalInclude   "\<include\>[?!]\@!" display
   syn keyword crystalInclude   extend require
   syn keyword crystalKeyword   caller typeof pointerof sizeof instance_sizeof
-  syn match   crystalRecord    "\<record\>[?!]\@!" display
+  syn match   crystalRecord    "^\s*\<record\>[?!]\@!\%(\s\w\)\@=" display
```

Comment:
I think it's hard to remove all false positives for this. However, as far as I checked the reference https://crystal-lang.org/api/0.21.0/toplevel.html#record%28name%2C%2Aproperties%29-macro, it seems that `record` must be followed by identifier which starts with upper character.

I feel it's better to get rid of many false positives. So I want to suggest:

```vim
syn match   crystalRecord    "\<record\>\%(\s\+\u\w*\)\@=" display
```

What do you think?

Reply:
Agree, that would be more precise.
Feel free to apply the improvement, or let me know if you want me to do that.